packages/pytest-simcore/src/pytest_simcore/openapi_specs.py

```diff
@@ -0,0 +1,104 @@
+# pylint: disable=redefined-outer-name
+# pylint: disable=unused-argument
+# pylint: disable=unused-variable
+# pylint: disable=too-many-arguments
+
+from collections.abc import Callable
+from copy import deepcopy
+from functools import lru_cache
+from pathlib import Path
+from typing import Any, NamedTuple
+
+import jsonref
+import pytest
+import yaml
+
+try:
+    from aiohttp import web
+
+    has_aiohttp = True
+except ImportError:
+    has_aiohttp = False
+
+
+class Entrypoint(NamedTuple):
+    name: str
+    method: str
+    path: str
+
+
+@lru_cache  # ANE: required to boost tests speed, gains 3.5s per test
+def _load(file: Path, base_uri: str = "") -> dict:
+    # SEE https://jsonref.readthedocs.io/en/latest/#lazy-load-and-load-on-repr
+    data: dict = jsonref.replace_refs(  # type: ignore
+        yaml.safe_load(file.read_text()),
+        base_uri=base_uri,
+        lazy_load=True,  # this data will be iterated
+        merge_props=False,
+    )
+    return data
+
+
+@pytest.fixture
+def openapi_specs_path() -> Path:
+    pytest.fail(reason="Must be overriden in caller tests package")
+
+
+@pytest.fixture
+def openapi_specs(openapi_specs_path: Path) -> dict[str, Any]:
+    assert openapi_specs_path.is_file()
+    openapi: dict[str, Any] = _load(
+        openapi_specs_path, base_uri=openapi_specs_path.as_uri()
+    )
+    return deepcopy(openapi)
+
+
+@pytest.fixture
+def openapi_specs_entrypoints(
+    openapi_specs: dict,
+) -> set[Entrypoint]:
```

Comment:
Q: why these fixtures are not session scoped? how are these used? cause the whole thing with needing to cache _load shows that this is probably an issue.

Reply:
The caching was already there. It was added by @GitHK with a note saying that it would increase substantially speed of tests so i left as it was

It is true that before we were loading with `openapi-core` and that part was dropped. Let me check if the cache is necessary.

I will also do them session scope.